ably/http/http.py

```
@@ -188,6 +188,11 @@
 
         hosts = self.get_rest_hosts()
         for retry_count, host in enumerate(hosts):
+            def should_stop_retrying():
+                time_passed = time.time() - requested_at
+                # if it's the last try or cumulative timeout is done, we stop retrying
+                return retry_count == len(hosts) - 1 or time_passed > http_max_retry_duration
```

Ruff (0.8.2)

Function definition does not bind loop variable `retry_count`

(B023)